app/src/main/java/com/hedvig/app/feature/genericauth/GenericAuthViewModel.kt

```diff
@@ -44,56 +29,56 @@ class GenericAuthViewModel(
   }
 
   fun setInput(value: String) {
-    _viewState.update { previousState ->
-      previousState.copy(
+    _viewState.update {
+      it.copy(
         input = value,
-        dirty = true,
-        error = computeError(previousState, value),
+        error = null,
       )
     }
   }
 
   fun clear() {
-    _viewState.update { it.copy(input = "", error = null, dirty = false, touched = false) }
-  }
-
-  fun blur() {
-    _viewState.update { previousState ->
-      val newState = previousState.copy(touched = previousState.dirty)
-      val error = computeError(newState, newState.input)
-      newState.copy(error = error)
-    }
+    _viewState.update { ViewState() }
   }
 
   fun submitEmail() {
-    viewModelScope.launch {
-      // TODO: Set Loading-state, once one exists
-      val email = viewState.value.input
-      when (val result = createOtpAttemptUseCase.invoke(email)) {
-        is CreateOtpAttemptUseCase.Result.Success -> {
-          eventChannel.trySend(Event.SubmitEmailSuccess(result.id, email))
-        }
-        CreateOtpAttemptUseCase.Result.Error -> {
-          _viewState.update { it.copy(error = ViewState.TextFieldError.NETWORK_ERROR) }
+    with(_viewState) {
```

Comment:
Hmm maybe I'm wrong on this one, but I find it kinda hard to follow through the code when we wrap things like this in a with() and use it as an implicit `this`.
I like `with` when we use it to bring it into scope since we're using things that need this thing to be a `this` to make extension functions work, like we did [here](https://github.com/HedvigInsurance/android/blob/0b31c97c8cf78d22abe7e0cee1ee0fcad6c2f6fe/app/src/main/java/com/hedvig/app/feature/marketing/marketpicked/MarketPickedScreen.kt#L39-L40) since .toDp() needs to be in the context of LocalDensity to work.

If you disagree I am 100% fine leaving it as it is, just sharing my opinion here 😊

Reply:
I see your point. Changing 👍